packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -604,4 +617,55 @@ contract PluginSetupProcessor is DaoAuthorizable {
             });
         }
     }
+
+    /// @notice Checks the update parameters
+    /// @param _updateParams The parameters of the update.
+    /// @return oldVersion The old semantic version number being needed in the context for further usage.
+    function _checkUpdateValidity(PluginUpdateParams calldata _updateParams)
```

Comment:
I agree that this is very ugly. However, the proper way
```
function _checkUpdateValidity(uint16[3] memory oldVersion, uint16[3] memory newVersion )
``` 
is not possible because of `stack too deep` errors. For the above to work, `newVersion` would have to be set in `prepareUpdate`, but we already have too many local variables (because of the event-based design instead of direct on-chain storage).

Reply:
With your current updates, now, I copied/pasted the code from `_checkUpdateValidity` into the `prepareUpdate` and no error about stack too deep. We could change the `_checkUpdateValidity` function name or put its code diretly into prepareUpdate. Passing the `oldVersion` and `newVersion` only don't make sense as well ! This is all true if we don't need this function anywhere.

Reply:
Removed it